### ISSUE TITLE
Ara_Broker_Reputations.lua:978: attempt to index field '?' (a nil value)

### ISSUE THREAD

**Issue by Ookami313** [AraBroker/Ara_Broker_Reputations#12]
Since the update to 10.2.0 I've been receiving the following error when I log into the game. Thanks.

2x ...ns/Ara_Broker_Reputations/Ara_Broker_Reputations.lua:978: attempt to index field '?' (a nil value)
[string "@Ara_Broker_Reputations/Ara_Broker_Reputations.lua"]:978: in function <...ns/Ara_Broker_Reputations/Ara_Broker_Reputations.lua:976>
[string "=(tail call)"]: ?

Locals:
self = AraReputation {
 0 = <userdata>
 PLAYER_ENTERING_WORLD = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:1184
 OnBackdropLoaded = <function> defined @SharedXML/Backdrop.lua:152
 GetBackdropColor = <function> defined @SharedXML/Backdrop.lua:390
 SetupTextureCoordinates = <function> defined @SharedXML/Backdrop.lua:214
 OnBackdropSizeChanged = <function> defined @SharedXML/Backdrop.lua:182
 HasBackdropInfo = <function> defined @SharedXML/Backdrop.lua:278
 SetBackdropBorderColor = <function> defined @SharedXML/Backdrop.lua:422
 SetupConfigMenu = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:982
 GetEdgeSize = <function> defined @SharedXML/Backdrop.lua:188
 ApplyBackdrop = <function> defined @SharedXML/Backdrop.lua:294
 SetBackdrop = <function> defined @SharedXML/Backdrop.lua:329
 ClearBackdrop = <function> defined @SharedXML/Backdrop.lua:282
 MAJOR_FACTION_RENOWN_LEVEL_CHANGED = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:976
 GetBackdropBorderColor = <function> defined @SharedXML/Backdrop.lua:409
 GetBackdrop = <function> defined @SharedXML/Backdrop.lua:347
 CHAT_MSG_COMBAT_FACTION_CHANGE = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:932
 SetupPieceVisuals = <function> defined @SharedXML/Backdrop.lua:246
 SetBackdropColor = <function> defined @SharedXML/Backdrop.lua:399
 AddModule = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:114
 SetBorderBlendMode = <function> defined @SharedXML/Backdrop.lua:266
}
factionId = 2574
newRenownLevel = 1
oldRenownLevel = 0
data = <table> {
 unlockDescription = "Complete the quest Home in the Dream by the Central Encampment in the Emerald Dream"
 renownReputationEarned = 0
 bountySetID = 119
 renownLevel = 1
 isUnlocked = true
 factionID = 2574
 expansionID = 9
 celebrationSoundKit = 213204
 name = "Dream Wardens"
 renownFanfareSoundKitID = 213208
 uiPriority = 6
 textureKit = "Dream"
 renownLevelThreshold = 2500
}
(*temporary) = nil
(*temporary) = <table> {
 start = 0
 max = 2500
}
(*temporary) = 2500
(*temporary) = "attempt to index field '?' (a nil value)"
GetMajorFactionData = <function> defined =[C]:-1
sessionStartMajorFaction = <table> {
 2511 = <table> {
 }
 2507 = <table> {
 }
 2564 = <table> {
 }
 2503 = <table> {
 }
 2510 = <table> {
 }
}
UpdateBar = <function> defined @Ara_Broker_Reputations/Ara_Broker_Reputations.lua:777

Also, is Ara Broke Reputations Alt ever been updated, or was it abandoned? It has been working fine for the most part up until the recent patch and now it's giving the following error when I log in to the game. Thanks.

3x ...ker_Reputations_Alts/Ara_Broker_Reputations_Alts.lua:85: table index is nil
[string "@Ara_Broker_Reputations_Alts/Ara_Broker_Reputations_Alts.lua"]:85: in function <...ker_Reputations_Alts/Ara_Broker_Reputations_Alts.lua:80>
[string "=(tail call)"]: ?

Locals:
(for index) = 93
(for limit) = 94
(for step) = 1
i = 93
name = nil
showValue = nil
level = nil
minVal = nil
maxVal = nil
value = nil
atWar = nil
canBeAtWar = nil
isHeader = nil
isCollapsed = nil
hasRep = nil
isWatched = nil
isChild = nil
(*temporary) = <table> {
 Ratchet = 16336
 Uldum Accord = 42000
 The Ascended = 42000
 The Avowed = 4681
 Gelkis Clan Centaur = 2806
 The Enlightened = 42000
 Waveblade Ankoan = 42000
 Corbyn = 42000
 Wrynn's Vanguard = 2400
 Valarjar = 42000
 The Undying Army = 42000
 Rajani = 42000
 Bizmo's Brawlpub = 1000
 Argussian Reach = 42000
 Bloodsail Buccaneers = -38125
 Brood of Nozdormu = 11368
 Defender Illona = 20000
 Alliance Vanguard = 42000
 Storm's Wake = 42000
 Everlook = 15727
 Syndicate = -10275
 Talonpriest Ishaal = 20000
 Akule Riverhorn = 42000
 Nat Pagle = 42000
 Court of Night = 42000
 The Nightfallen = 42000
 Bizmo's Brawlpub (Season 3) = 3000
 Valdrakken Accord = 3000
 The Wardens = 42000
 Avengers of Hyjal = 13168
 Army of the Light = 42000
 The Tillers = 42000
 Sha'leth = 42000
 The Wild Hunt = 42000
 Obsidian Warders = 20275
 Maruuk Centaur = 3000
 Hydraxian Waterlords = 29074
 Shen'dralar = 200
 Bizmo's Brawlpub (Season 2) = 5000
 Court of Harvesters = 42000
 Tormmok = 20000
 Delvar Ironfist = 20000
 Court of Farondis = 42000
 Sabellian = 25200
 Death's Advance = 42000
 The League of Arathor = 3450
 Ve'nari = 42000
 Stitchmasters = 21000
 Champions of Azeroth = 42000
 Dragonscale Expedition = 3000
 Impus = 42000
 Proudmoore Admiralty = 42000
 Iskaara Tuskarr = 3000
 The Scryers = -42000
 Emperor Shaohao = 42000
 Leorajh = 20000
 Wrathion = 25200
 Booty Bay = 17047
 Tortollan Seekers = 42000
 Soridormi = 42000
 Rustbolt Resistance = 42000
 The Anglers = 42000
 Armies of Legionfall = 42000
 Silverwing Sentinels = 1920
 Gadgetzan = 17352
 Magram Clan Centaur = -5342
 Ilyssia of the Waters = 42000
 The Archivists' Codex = 41000
 Dreamweavers = 42000
 Highmountain Tribe = 42000
 Artisan's Consortium - Dragon Isles Branch = 12500
 Cobalt Assembly = 3600
 Keeper Raynae = 42000
 Ravenholdt = 55
 Order of Embers = 42000
 Loamm Niffen = 3000
 7th Legion = 42000
}
(*temporary) = nil
(*temporary) = "table index is nil"
info = <table> {
}
char = <table> {
 class = "HUNTER"
 factions = <table> {
 }
 faction = "Alliance"
}

**Comment by mightykong** [comment]
I am not the original author, I've just been updating Ara Broker Reputations.  As far as Ara Broker Reputations Alt, I've never looked at it and never updated it.  I'd consider it abandoned.

As to the error, I did get an error when I've accepted quests that reference the new Reputation on before I'd unlocked it on them.  I'm looking into that, but don't have a fix yet.  Also, I would try disabling Ara Broker Reputations Alt, and try loading again.  If it gives the error, do a /console reloadui and see if it clears the error. 

Report back if you get any more data after trying the reload and I'll work on digging into this error over the next day or so
